src/host/proxy/ITerminalHandoff.idl

```diff
@@ -4,15 +4,24 @@
 import "oaidl.idl";
 import "ocidl.idl";
 
+
+typedef struct _TERMINAL_STARTUP_INFO
+{
+    LPCWSTR pszTitle;
+    LPCWSTR pszIconPath;

```

Comment:
Who owns these strings? Does RPC manage their lifetime? Should they be `BSTR`s? When do they get deleted?

Reply:
(Some random SO post says "If you have any control whatsoever, use `BSTR`, fwiw)

Reply:
I literally have no idea how this works and I was having a hard time finding docs. It did seem to, but I suspect they're owned by OpenConsole, which like _what_.

Reply:
> Who owns these strings? Does RPC manage their lifetime? Should they be `BSTR`s? When do they get deleted?

I believe these parameters got `[string]` attributes by default by the MIDL compiler. This means that the lifetime and transmission is transparently handled by the runtime and so their use here is fine and safe.
But considering that the `BSTR` types are specifically made for COM usage, we might want to use them anyways. They're length-prefixed too, so that's another nice bonus. But apart from that, `LPCWSTR` works fine IMO.